### PR TITLE
[routing] 50m threshold to skip next turn distance notification.

### PR DESCRIPTION
**Pull request by vng**
Thanks @Shura0 for the initial patch here https://github.com/organicmaps/organicmaps/pull/4354

### CODE REVIEW

File: routing/turns_notification_manager.cpp [hunk 1]
```diff
@@ -1,18 +1,24 @@
...
+namespace sound
+{
+
 namespace
 {
 // If the distance between two sequential turns is less than kMaxTurnDistM
 // the information about the second turn will be shown or pronounced when the user is
 // approaching to the first one.
 double constexpr kMaxTurnDistM = 400.0;
+double constexpr kMinTurnDistM = 50.0;
```
Comment (biodranik) [inline routing/turns_notification_manager.cpp:21]: Need a comment on that constant too.

File: routing/turns_tts_text.cpp [hunk 2]
```diff
@@ -54,17 +54,25 @@ std::string GetTtsText::GetTurnNotification(Notification const & notification) c
...
   std::string const dirStr = GetTextById(GetDirectionTextId(notification));
   if (dirStr.empty())
-    return std::string{};
+    return {};
 
-  std::string const distStr = GetTextById(GetDistanceTextId(notification));
+  std::string distStr;
+  if (notification.m_useThenInsteadOfDistance)
+  {
+    distStr = GetTextById("then");
```
Comment (biodranik) [inline routing/turns_tts_text.cpp:70]: Should a comma be inserted here before "then"? Like ", then"?

File: routing/turns_tts_text.cpp [hunk 3]
```diff
@@ -97,8 +105,8 @@ std::string GetTtsText::GetTextById(std::string const & textId) const
 
 std::string GetDistanceTextId(Notification const & notification)
 {
-  if (notification.m_useThenInsteadOfDistance)
-    return "then";
+//  if (notification.m_useThenInsteadOfDistance)
```
Comment (biodranik) [inline routing/turns_tts_text.cpp:108]: Remove unused parameter?

**Review by biodranik — APPROVED** [review]
LGTM if someone tests it and confirms that it works.

File: routing/turns_tts_text.cpp (hunk 2 above)
Comment (vng) [reply routing/turns_tts_text.cpp:70]: The notation is defined for tts, no any commas needed.

File: routing/turns_tts_text.cpp (hunk 3 above)
Comment (vng) [reply routing/turns_tts_text.cpp:108]: It is used.

File: routing/routing_tests/turns_tts_text_tests.cpp [hunk 4]
```diff
@@ -24,9 +24,9 @@ UNIT_TEST(GetDistanceTextIdMetersTest)
   Notification const notification1(500, 0, false, CarDirection::TurnRight,
                                    measurement_utils::Units::Metric);
   TEST_EQUAL(GetDistanceTextId(notification1), "in_500_meters", ());
-  Notification const notification2(500, 0, true, CarDirection::TurnRight,
-                                   measurement_utils::Units::Metric);
-  TEST_EQUAL(GetDistanceTextId(notification2), "then", ());
+//  Notification const notification2(500, 0, true, CarDirection::TurnRight,
```
Comment (biodranik) [inline routing/routing_tests/turns_tts_text_tests.cpp:27]: А с этими тестами что?

**Review by biodranik — APPROVED** [review]

File: routing/routing_tests/turns_tts_text_tests.cpp (hunk 4 above)
Comment (vng) [reply routing/routing_tests/turns_tts_text_tests.cpp:27]: Функция GetDistanceTextId больше не отвечает за "then".

**Pull request merged**